src/services/review/RepoCheckerService.ts

```diff
@@ -77,33 +78,57 @@ export default class RepoCheckerService {
     this.pageService = pageService
   }
 
-  isCurrentlyLocked(repo: string): ResultAsync<true, SiteCheckerError> {
+  createLogsFolder(
+    repo: string
+  ): ResultAsync<string | undefined, SiteCheckerError> {
     const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
     // create logs folder if it does not exist
+    return ResultAsync.fromPromise(doesDirectoryExist(logsFilePath), (err) => {
+      logger.error(
+        `SiteCheckerError: error finding if logs directory exists for ${repo} :${err}`
+      )
+      return new SiteCheckerError(
+        `Error finding if logs directory exists for ${repo} :${err}`
+      )
+    }).andThen((doesLogsFolderExist) => {
+      if (!doesLogsFolderExist) {
+        return ResultAsync.fromPromise(
+          fs.promises.mkdir(logsFilePath, { recursive: true }),
+          (err) => {
+            logger.error(`SiteCheckerError: Unable to create logs dir: ${err}`)
+            return new SiteCheckerError(
+              `SiteCheckerError: Unable to create logs dir: ${err}`
+            )
+          }
+        )
+      }
+      return okAsync("ok")
+    })
+  }
 
-    if (!fs.existsSync(logsFilePath)) {
-      fs.mkdirSync(logsFilePath, { recursive: true })
-    }
-
+  isCurrentlyLocked(repo: string): ResultAsync<true, SiteCheckerError> {
+    const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
     // check if checker.lock exists
     const lockFilePath = path.join(logsFilePath, LOCK_CONTENT)
-
-    return ResultAsync.fromPromise(
-      fs.promises.access(lockFilePath, fs.constants.F_OK),
-      (err) => new SiteCheckerError(`${err}`)
-    ).map(() => true)
+    return this.createLogsFolder(repo)
+      .andThen(() =>
+        ResultAsync.fromPromise(
+          fs.promises.access(lockFilePath, fs.constants.F_OK),
+          (err) => new SiteCheckerError(`${err}`)
+        )
+      )
+      .map(() => true)
```

Comment:
double checking - we're checking if we can access the newly created lock file right? in the event that we can't, should we also perform some clean-up? (we could file a ticket for this, no worri3es)

Reply:
nope, `this.createLogsFolder` is just a sanity check that the top level folder exists. 
the key thing that this function does is check if the `checker.lock` exists, which is done by the `fs.promise.access` call. 

the lock file creation is done in a separate function call. the clean up functionality already exists (if some error exists, create a error file + remove the `checker.lock` file)

Reply:
hmm what i'm hearing is that the point of `isCurrentlyLocked` is the check on `checker.lock`! seems abit weird to have the log creation folder here then? we can probably shift it out and this would be abit less confusing. (since this PR is scoped to fixing `fsSync`, we can do it later)

Reply:
the dir structure is 
-  pa-corp (folder)
   - logs.csv (file)
   - checker.lock (file)

what we are doing here is ensuring that the top level folder exists before checking for the checker.lock ya